cve_bin_tool/extractor.py

```diff
@@ -43,23 +49,47 @@ def __init__(self, logger=None, error_mode=ErrorMode.TruncTrace):
         self.logger = logger or LOGGER.getChild(self.__class__.__name__)
         self.error_mode = error_mode
         self.tempdir = None
+        # Adding filetype LZMA (see comments on line 438)
+        filetype.add_type(Lzma())
         self.file_extractors = {
-            self.extract_file_tar: {".tgz", ".tar.gz", ".tar", ".tar.xz", ".tar.bz2"},
-            self.extract_file_rpm: {".rpm"},
-            self.extract_file_deb: {".deb", ".ipk"},
-            self.extract_file_cab: {".cab"},
-            self.extract_file_apk: {".apk"},
-            self.extract_file_zst: {".zst"},
-            self.extract_file_pkg: {".pkg"},
+            self.extract_file_tar: {
+                EXTENSIONS: [
+                    ".tgz",
+                    ".tar.gz",
+                    ".tar",
+                    ".tar.xz",
+                    ".tar.bz2",
+                    ".xz",
+                    ".bz2",
+                    ".gz",
+                ],
+                MIMES: [],
+            },
+            self.extract_file_rpm: {EXTENSIONS: [".rpm"], MIMES: []},
+            self.extract_file_deb: {EXTENSIONS: [".deb", ".ipk"], MIMES: []},
+            self.extract_file_cab: {EXTENSIONS: [".cab"], MIMES: []},
+            self.extract_file_apk: {EXTENSIONS: [".apk"], MIMES: []},
+            self.extract_file_zst: {EXTENSIONS: [".zst"], MIMES: []},
+            self.extract_file_pkg: {EXTENSIONS: [".pkg"], MIMES: []},
             self.extract_file_zip: {
-                ".exe",
-                ".zip",
-                ".jar",
-                ".msi",
-                ".egg",
-                ".whl",
-                ".war",
-                ".ear",
+                EXTENSIONS: [
+                    ".exe",
+                    ".zip",
+                    ".jar",
+                    ".msi",
+                    ".egg",
+                    ".whl",
+                    ".war",
+                    ".ear",
+                ],
+                MIMES: [
+                    "application/x-msdownload",
+                    "application/x-7z-compressed",
+                    "application/x-lzip",
+                    "application/lzma",
+                    "application/x-tar",
+                    "appication/gzip",
```

Comment:
i know it's a first draft, but in looking at the different types, it seems like the mimes for `7z`, `x-tar`, `gzip`, and others might fit better under the `extract_file_tar` mimes

thoughts?

Reply:
I think it makes sense for `x-tar` and `gzip`, maybe `7z` fits better under `zip`?